Sources/SwiftParser/Declarations.swift

```diff
@@ -1120,10 +1120,39 @@ extension Parser {
     _ handle: RecoveryConsumptionHandle
   ) -> RawDeinitializerDeclSyntax {
     let (unexpectedBeforeDeinitKeyword, deinitKeyword) = self.eat(handle)
+    
     var unexpectedNameAndSignature: [RawSyntax?] = []
-    unexpectedNameAndSignature.append(self.consume(if: TokenSpec(.identifier, allowAtStartOfLine: false)).map(RawSyntax.init))
-    if self.at(.leftParen) && !self.currentToken.isAtStartOfLine {
-      unexpectedNameAndSignature.append(RawSyntax(parseFunctionSignature()))
+    var unexpectedAfterAsync: [RawSyntax?] = []
+
+    // async is a contextual keyword
+    // must be parsed before attempting to parse identifier
+    var effectSpecifiers = parseDeinitEffectSpecifiers()
+    
+    if effectSpecifiers == nil {
+      if let identifier = self.consume(if: TokenSpec(.identifier, allowAtStartOfLine: false)).map(RawSyntax.init) {
+        unexpectedNameAndSignature.append(identifier)
+      }
+      effectSpecifiers = parseDeinitEffectSpecifiers()
+    }
+    if effectSpecifiers == nil && self.at(.leftParen) && !self.currentToken.isAtStartOfLine {
+      let sig = parseFunctionSignature()
+      unexpectedNameAndSignature.append(contentsOf: sig.unexpectedBeforeInput?.elements ?? [])
+      unexpectedNameAndSignature.append(RawSyntax(sig.input))
+      unexpectedNameAndSignature.append(contentsOf: sig.unexpectedBetweenInputAndEffectSpecifiers?.elements ?? [])
+      if let effects = sig.effectSpecifiers {
+        effectSpecifiers = RawDeinitEffectSpecifiersSyntax(
+          effects.unexpectedBeforeAsyncSpecifier,
+          asyncSpecifier: effects.asyncSpecifier,
+          effects.unexpectedBetweenAsyncSpecifierAndThrowsSpecifier,
+          throwsSpecifier: effects.throwsSpecifier,
+          effects.unexpectedAfterThrowsSpecifier,
+          arena: self.arena
+        )
+      }
+      unexpectedAfterAsync.append(contentsOf: sig.unexpectedBetweenEffectSpecifiersAndOutput?.elements ?? [])
+      if let output = sig.output {
+          unexpectedAfterAsync.append(RawSyntax(output))
```

Comment:
Incorrect  indentation. Could you run `./format.py`?

Reply:
Sure. Is it ok to run it once as a separate commit, or should I patch every commit?